gap/base/recognition.gi

```diff
@@ -528,6 +528,7 @@ InstallGlobalFunction( RecogniseGeneric,
               counter,").");
         fi;
         Add(depth,'F');
+        Assert(2, ForAll(GeneratorsOfGroup(H), x->ValidateHomomInput(ri, x)));
```

Comment:
There is another call to `ImageElm` around line 631, which also should at the very least get such an assertion.